rasa_nlu_examples/classifiers/sparse_naive_bayes_intent_classifier.py

```diff
@@ -47,215 +26,14 @@ def required_components(cls) -> List[Type[Component]]:
         "class_prior": None,
     }
 
-    def __init__(
-        self,
-        component_config: Optional[Dict[Text, Any]] = None,
-        clf: Optional["sklearn.naive_bayes.BernoulliNB"] = None,
-        le: Optional["sklearn.preprocessing.LabelEncoder"] = None,
-    ) -> None:
-        """Construct a new naive Bayes intent classifier using the sklearn framework."""
-        from sklearn.preprocessing import LabelEncoder
-
-        super().__init__(component_config)
-
-        if le is not None:
-            self.le = le
-        else:
-            self.le = LabelEncoder()
-        self.clf = clf
-
-    @classmethod
-    def required_packages(cls) -> List[Text]:
-        return ["sklearn"]
-
-    def transform_labels_str2num(self, labels: List[Text]) -> np.ndarray:
-        """
-        Transforms a list of strings into numeric label representation.
-
-        :param labels: List of labels to convert to numeric representation
-        :returns: numpy array of numeric label ids.
-        """
-
-        return self.le.fit_transform(labels)
-
-    def transform_labels_num2str(self, y: np.ndarray) -> np.ndarray:
-        """
-        Transforms a numpy array of numeric label ids into a list of string label ids.
+    def create_sklearn_classifier(
+        self, **kwargs: Any
+    ) -> "sklearn.base.ClassifierMixin":
+        r"""Lazily imports the required sklearn classifier class and creates and
+        instantiates the sklearn classifier using all the given keyword arguments.
 
-        :param y: array of labels to convert to string representation
-        :returns: an ndarray of label id strings
+        :param **kwargs: see defaults dictionary
         """
-
-        return self.le.inverse_transform(y)
-
-    def train(
-        self,
-        training_data: TrainingData,
-        config: Optional[RasaNLUModelConfig] = None,
-        **kwargs: Any,
-    ) -> None:
-        """Train the intent classifier on a data set."""
-
         from sklearn.naive_bayes import BernoulliNB
```

Comment:
I'm curious, why not load the `BernoulliNB` on top of the file?

Reply:
I had just assumed that sklearn was an optional dependency and hence we'd run into problems if just importing it on top of the file. The previous BernoulliNB class did that, as well as the SklearnIntentClassifier aka SVC (https://github.com/RasaHQ/rasa/blob/main/rasa/nlu/classifiers/sklearn_intent_classifier.py). If  that's no problem, then I'll move those imports.

Reply:
In Rasa `sklearn` is a mandatory dependency. Without it the countvectoriser would break as well as some of our evaluation and train/test splitting scripts. I'd be fine with moving it up 👍